### ISSUE TITLE
Stuttering sound when using set_looping(true)

### ISSUE THREAD

**Issue by eliukblau** [jhasse/ears#1]
Hello.

I'm having trouble with the `set_looping` function from `Music`. Having the following snippet:

```
extern crate ears;

use ears::{Music, AudioController};

fn main() {
    let mut music = Music::new("res/music.ogg").unwrap();
    // This start the stutter:
    music.set_looping(true);
    music.play();
    while music.is_playing() {}
}
```

When run, it starts stuttering the first milliseconds of the sound forever. Not using `set_looping` makes the sound play properly as expected (only once). This didn't happened with the provided `explosion.ogg` file so I pressume the problem arises when dealing with long (> 2min) files.

I'm using:
- OSX 10.9.5
- Homebrew:
  - openal-soft 1.17.0
  - libsndfile: 1.0.26
- Rustc: 1.6.0

**Comment by ramn** [comment]
Same issue for me, Ubuntu 15.10, rustc 1.6.0

**Comment by jhasse** [comment]
It seems to me that looping Music was never properly implemented.

Can you test if it works now? I'll upload 0.3.2 to crates.io in a moment.

**Comment by eliukblau** [comment]
It works perfectly on OSX now! :D Thank you!

**Comment by ramn** [comment]
If I drop the Music object and recreate it, now I get:

```
OpenAL error : Illegal AL call.
thread '<main>' panicked at 'called `Option::unwrap()` on a `None` value', ../src/libcore/option.rs:367
```

**Comment by ramn** [comment]
Actually, I probably got that before this fix aswell. Looping works now, thanks!

**Comment by jhasse** [comment]
That's good to hear :)

Can you create a new issue about the illegal AL call? Thanks!

**Comment by ramn** [comment]
Done!
